File: src/MidnightOil.java
/**
 * Created by kapilkrishnakumar on 9/30/15.
 */
//http://codeforces.com/problemset/problem/165/B/
import java.util.*;
public class MidnightOil {

    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int k = sc.nextInt();

        int low = 0;
        int high = n;
        int mid = 0;
        while( low <= high){
            mid = low + (high - low) / 2;
            boolean check = check(n, mid, k);
            if(check && !check(n , mid - 1, k))
                break;
            if(check)
                high = mid - 1;
            else
                low = mid + 1;
        }
        System.out.println(mid);
    }

    public static boolean check(int n, int v , int k){
        int curr = v;
        int sum = 0;
        while(curr != 0){
            sum += curr;
            curr /= k;
        }
        return sum >= n;
    }
}

File: Hackerrank/Python/midnightoil.py
def main():
	val = input()
	valS = val.split(' ')
	n = int(valS[0])
	k = int(valS[1])

	lower = 0
	high = n
	while lower <= high:
		printf("mid")
		mid = lower + (high - lower)/2
		if check(n, mid, k):
			high = mid - 1
		else:
			lower = mid + 1

	print(lower)

def check(n , v , k):
	denom = 1;
	sum = 0
	while (v / denom) != 0:
		print(v / denom)
		denom *= k
		sum += int(v / denom)
	return sum >= n


if __name__ == '__main__':
	main()